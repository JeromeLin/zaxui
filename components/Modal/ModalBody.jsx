import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class ModalBody extends PureComponent {

  render() {
    const { prefixCls, className, height, children, ...others } = this.props;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
    });

    const bodyStyle = {
      height,
    };

    return (
      <div className={cls} style={bodyStyle} {...others}>
        {children}
      </div>
    );
  }
}

ModalBody.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

ModalBody.defaultProps = {
  prefixCls: 'za-modal-body',
  className: null,
};

export default ModalBody;

