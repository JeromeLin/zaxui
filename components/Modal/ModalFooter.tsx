import React, { PureComponent } from 'react';
import classnames from 'classnames';

export interface ModalFooterProps {
  prefixCls?: string;
  className?: string;
}

export default class ModalFooter extends PureComponent<ModalFooterProps, {}> {

  static defaultProps = {
    prefixCls: 'za-modal',
  };

  render() {
    const { prefixCls, className, children, ...others } = this.props;
    const cls = classnames(`${prefixCls}-footer`, className, {
      // block: true,
    });

    return (
      <div className={cls} {...others}>
        {children}
      </div>
    );
  }
}
