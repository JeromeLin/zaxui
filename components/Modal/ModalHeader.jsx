import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Icon from '../Icon';

class ModalHeader extends PureComponent {

  render() {
    const { prefixCls, className, title, onClose, ...others } = this.props;

    const cls = classnames({
      [`${prefixCls}`]: true,
      [className]: !!className,
    });

    const btnClose = onClose
      ? <Icon type="wrong" className={`${prefixCls}-close`} onClick={onClose} />
      : null;

    return (
      <div className={cls} {...others}>
        <div className={`${prefixCls}-title`}>{title}</div>
        {btnClose}
      </div>
    );
  }
}

ModalHeader.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  title: PropTypes.string,
  onClose: PropTypes.func,
};

ModalHeader.defaultProps = {
  prefixCls: 'za-modal-header',
  title: '',
};

export default ModalHeader;

