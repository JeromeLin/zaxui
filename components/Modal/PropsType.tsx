interface BaseProps {
  shape?: 'radius';
  visible?: boolean;
  animationType?: 'fade' | 'door' | 'flip' | 'rotate' | 'zoom' | 'moveUp' | 'moveDown' | 'moveLeft' | 'moveRight' | 'slideUp' | 'slideDown' | 'slideLeft' | 'slideRight';
  animationDuration?: number;
  width?: string | number;
  onMaskClick?: () => void; 
}

export interface ModalProps extends BaseProps {
  prefixCls?: string;
  className?: string;
}

interface ModalHeaderBaseProps extends BaseProps {
  title?: any;
  onClose?: () => void;
}

export interface ModalHeaderProps extends ModalHeaderBaseProps {
  prefixCls?: string;
  className?: string;
}

interface ModalBodyBaseProps extends BaseProps {
  height?: string | number;
}

export interface ModalBodyProps extends ModalBodyBaseProps {
  prefixCls?: string;
  className?: string;
}

export interface ModalFooterProps {
  prefixCls?: string;
  className?: string;
}