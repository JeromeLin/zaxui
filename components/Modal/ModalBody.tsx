import React, { PureComponent, CSSProperties } from 'react';
import classnames from 'classnames';
import { ModalBodyProps } from './PropsType';

export { ModalBodyProps };

export default class ModalBody extends PureComponent<ModalBodyProps, {}> {

  static defaultProps = {
    prefixCls: 'za-modal',
  };

  render() {
    const { prefixCls, className, height, children, ...others } = this.props;
    const cls = classnames(`${prefixCls}-body`, className);

    const bodyStyle: CSSProperties = {};
    bodyStyle.height = height;

    return (
      <div className={cls} style={bodyStyle} {...others}>
        {children}
      </div>
    );
  }
}
