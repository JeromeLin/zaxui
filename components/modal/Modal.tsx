import React, { Component } from 'react';
import classnames from 'classnames';
import { BaseModalProps } from './PropsType';
import Popup from '../popup';
import ModalHeader from './ModalHeader';
import ModalBody from './ModalBody';
import ModalFooter from './ModalFooter';
import Trigger from '../trigger';

export interface ModalProps extends BaseModalProps {
  prefixCls?: string;
  className?: string;
}

export default class Modal extends Component<ModalProps, any> {
  static alert;

  static confirm;

  static defaultProps: ModalProps = {
    prefixCls: 'za-modal',
    visible: false,
    animationType: 'fade',
    animationDuration: 200,
    width: '70%',
    mask: true,
    maskType: 'normal',
    shape: 'radius',
    closable: false,
    maskClosable: false,
    destroy: true,
  };

  onClose = () => {
    const { onCancel } = this.props;
    onCancel && onCancel();
  };

  render() {
    const { visible, prefixCls, className, shape, children, mountContainer, maskClosable, title, closable, footer, onCancel, ...others } = this.props;

    const cls = {
      modal: classnames(prefixCls, className, {
        [`${prefixCls}--${shape}`]: !!shape,
      }),
      dialog: classnames(`${prefixCls}__dialog`),
    };

    const showHeader = title || closable;
    const noop = () => { };

    return (
      <Trigger visible={visible || false} onClose={this.onClose}>
        <Popup
          visible={visible}
          className={cls.modal}
          direction="center"
          onMaskClick={maskClosable ? onCancel : noop}
          mountContainer={mountContainer}
          {...others}
        >
          <div className={cls.dialog}>
            {showHeader && <ModalHeader title={title} closable={closable} onCancel={onCancel} />}
            <ModalBody>{children}</ModalBody>
            {footer && <ModalFooter>{footer}</ModalFooter>}
          </div>
        </Popup>
      </Trigger>
    );
  }
}
