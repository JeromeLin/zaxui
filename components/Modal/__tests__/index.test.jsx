import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Modal from '../index';

describe('Modal', () => {
  it('renders correctly', () => {
    const onMaskClick = jest.fn();
    const onClose = jest.fn();

    const wrapper = render(
      <Modal visible onMaskClick={onMaskClick}>
        <Modal.Header title="标题" onClose={onClose} />
        <Modal.Body>
          foo
        </Modal.Body>
      </Modal>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
