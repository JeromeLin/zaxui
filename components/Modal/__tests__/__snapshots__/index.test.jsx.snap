// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<Modal
  animationDuration={200}
  animationType="fade"
  className={null}
  onMaskClick={[Function]}
  prefixCls="za-modal"
  shape={null}
  visible={true}
  width="70%"
>
  <div
    className="za-modal"
    style={
      Object {
        "WebkitAnimationDuration": "200ms",
        "animationDuration": "200ms",
      }
    }
  >
    <div
      className="za-modal-wrapper"
    >
      <div
        className="za-modal-dialog"
        onClick={[Function]}
        style={
          Object {
            "WebkitAnimationDuration": "200ms",
            "animationDuration": "200ms",
            "width": "70%",
          }
        }
      >
        <ModalHeader
          className={null}
          onClose={[Function]}
          prefixCls="za-modal-header"
          title="标题"
        >
          <div
            className="za-modal-header"
          >
            <div
              className="za-modal-header-title"
            >
              标题
            </div>
            <Icon
              className="za-modal-header-close"
              onClick={[Function]}
              prefixCls="za-icon"
              theme={null}
              type="wrong"
            >
              <i
                className="za-icon za-modal-header-close za-icon-wrong"
                onClick={[Function]}
              />
            </Icon>
          </div>
        </ModalHeader>
        <ModalBody
          className={null}
          prefixCls="za-modal-body"
        >
          <div
            className="za-modal-body"
            style={
              Object {
                "height": undefined,
              }
            }
          >
            foo
          </div>
        </ModalBody>
      </div>
    </div>
    <Mask
      className={null}
      onClose={[Function]}
      prefixCls="za-mask"
      type="normal"
      visible={true}
    >
      <div
        className="za-mask normal"
        onClick={[Function]}
      />
    </Mask>
  </div>
</Modal>
`;
